examples/connect.rs

```diff
@@ -186,48 +188,31 @@ mod udp {
         let udp = UdpSocket::bind(&addr_to_bind)
             .expect("failed to bind socket");
 
-        // Like above with TCP we use an instance of `UdpCodec` to transform
-        // this UDP socket into a framed sink/stream which operates over
-        // discrete values. In this case we're working with *pairs* of socket
-        // addresses and byte buffers.
-        let (sink, stream) = udp.framed(Bytes).split();
-
-        // All bytes from `stdin` will go to the `addr` specified in our
-        // argument list. Like with TCP this is spawned concurrently
-        pool.execute(stdin.map(move |chunk| {
-            (addr, chunk)
-        }).forward(sink).then(|result| {
-            if let Err(e) = result {
-                panic!("failed to write to socket: {}", e)
-            }
-            Ok(())
-        })).unwrap();
+        // Move the udp socket into an `Arc` so that it can be used from
+        // multiple tasks.
+        let rx = Arc::new(udp);
+        let tx = rx.clone();
+
+        pool.execute(stdin.for_each(move |chunk| {
+            // The `read_stdin` provides chunks small enough to fit in a UDP
+            // packet.
+            tx.send_to(&chunk, &addr)
+                .map(|_| ())
+        }).map_err(|_| ())).unwrap();
 
         // With UDP we could receive data from any source, so filter out
         // anything coming from a different address
-        Box::new(stream.filter_map(move |(src, chunk)| {
-            if src == addr {
-                Some(chunk.into())
-            } else {
-                None
-            }
-        }))
-    }
+        Box::new(stream::poll_fn(move || {
+            let mut buf = [0; 1024];
```

Comment:
1kb here, see below.

Reply:
This is not very scientific... this should probably be 1500 though as that is a common value to use as it is a pretty safe to assume MTU. While UDP supports up to 65kb, a lot of hops will have much smaller max packet sizes, so large packet sizes tend to drop.

The packet size is usually defined as part of the protocol being implemented on top of UDP, so since we are doing our own protocol, we can define the max packet size to be whatever we want.